HaxeManual/05-expressions.tex

```diff
@@ -169,11 +169,242 @@ \section{Constants}
+
+\paragraph{Compound assignment operators}
+
+\begin{center}
+\begin{tabular}{| l | l | l | l | l |}
+	\hline
+	Operator & Operation & Operand 1 & Operand 2 & Result type \\ \hline
+	\expr{\%=} & modulo & \type{Float/Int} & \type{Float/Int} & \type{Float/Int} \\
+	\expr{*=} & multiplication & \type{Float/Int} & \type{Float/Int} & \type{Float/Int} \\
+	\expr{/=} & division & \type{Float} & \type{Float/Int} & \type{Float} \\
+	\expr{+=} & addition & \type{Float/Int} & \type{Float/Int} & \type{Float/Int} \\
+	\expr{-=} & subtraction & \type{Float/Int} & \type{Float/Int} & \type{Float/Int} \\
+	\expr{<<=} & shift left & \type{Int} & \type{Int} & \type{Int} \\
+	\expr{>>=} & shift right & \type{Int} & \type{Int} & \type{Int} \\
+	\expr{>>>=} & unsigned shift right & \type{Int} & \type{Int} & \type{Int} \\
+	\expr{\&=} & bitwise and & \type{Int} & \type{Int} & \type{Int} \\
+	\expr{|=} & bitwise or & \type{Int} & \type{Int} & \type{Int} \\
+	\expr{\^=} & bitwise xor & \type{Int} & \type{Int} & \type{Int}
+\end{tabular}
+\end{center}
+
+In all cases, a compound assignment modifies the value of a variable, then returns the new value.
+
+\paragraph{Numeric comparison operators}
+
+\begin{center}
+\begin{tabular}{| l | l | l | l | l |}
+	\hline
+	Operator & Operation & Operand 1 & Operand 2 & Result type \\ \hline
+	\expr{==} & equal & \type{Float/Int} & \type{Float/Int} & \type{Bool} \\
+	\expr{!=} & not equal & \type{Float/Int} & \type{Float/Int} & \type{Bool} \\
+	\expr{<} & less than & \type{Float/Int} & \type{Float/Int} & \type{Bool} \\
+	\expr{<=} & less than or equal & \type{Float/Int} & \type{Float/Int} & \type{Bool} \\
+	\expr{>} & greater than & \type{Float/Int} & \type{Float/Int} & \type{Bool} \\
+	\expr{>=} & greater than or equal & \type{Float/Int} & \type{Float/Int} & \type{Bool}
+\end{tabular}
+\end{center}
+
+\paragraph{String comparison operators}
+
+\begin{center}
+\begin{tabular}{| l | l | l | l | l |}
+	\hline
+	Operator & Operation & Operand 1 & Operand 2 & Result type \\ \hline
+	\expr{==} & equal & \type{String} & \type{String} & \type{Bool} \\
+	\expr{!=} & not equal & \type{String} & \type{String} & \type{Bool} \\
+	\expr{<} & lexicographically before & \type{String} & \type{String} & \type{Bool} \\
+	\expr{<=} & lexicographically before or equal & \type{String} & \type{String} & \type{Bool} \\
+	\expr{>} & lexicographically after & \type{String} & \type{String} & \type{Bool} \\
+	\expr{>=} & lexicographically after or equal & \type{String} & \type{String} & \type{Bool}
+\end{tabular}
+\end{center}
+
+\type{String} is an immutable type in Haxe, so two separate instances of the same string are considered equal.
+
+\paragraph{Equality operators}
+
+\begin{center}
+\begin{tabular}{| l | l | l | l | l |}
+	\hline
+	Operator & Operation & Operand 1 & Operand 2 & Result type \\ \hline
+	\expr{==} & equal & any & same as operand 1 & \type{Bool} \\
```

Comment:
"same" is not accurate, it's fine as long as the assignment can be made. For instance `baseClass == childClass` works as expected.

Reply:
How about `any & any` in the table and a note saying that the types of operands 1 and 2 must unify (+ link to unification section)?